packages/x-charts/src/PieChart/PieArc.tsx

```diff
@@ -84,28 +76,18 @@ export default function PieArc(props: PieArcProps) {
     classes: innerClasses,
     color,
     highlightScope,
-    innerRadius: baseInnerRadius = 0,
-    outerRadius: baseOuterRadius,
-    cornerRadius: baseCornerRadius = 0,
-    highlighted,
-    faded = { additionalRadius: -5 },
     onClick,
+    isFaded,
+    isHighlighted,
+    startAngle,
+    endAngle,
+    paddingAngle,
+    innerRadius,
+    outerRadius,
+    cornerRadius,
     ...other
   } = props;
 
-  const getInteractionItemProps = useInteractionItemProps(highlightScope);
-
-  const { item } = React.useContext(InteractionContext);
-
-  const isHighlighted = getIsHighlighted(
-    item,
-    { type: 'pie', seriesId: id, dataIndex },
-    highlightScope,
-  );
-
-  const isFaded =
-    !isHighlighted && getIsFaded(item, { type: 'pie', seriesId: id, dataIndex }, highlightScope);
-
```

Comment:
All those stuff are moved in the parent.

`isHighlighted`, `isFaded` and the props added are added to data in the `useTransformData`